### ISSUE TITLE
Testing on real device

### ISSUE THREAD

**Issue by Lengo46** [ArtSabintsev/Siren#141]
Hi, love the framework.

I would like to submit this with my next App update. I've tested the popup, and it works on the simulator when I downcast my current build number against the (local) Appstore version (1.01). When i'm testing it on a real device, the popup keeps on showing, regardless how I downcast the build number. Something that i'm missing here?

Thanks in advanced.

**Comment by ArtSabintsev** [comment]
Hey!

Can you share some sample code of what how Siren is setup in your project? I have a feeling you're using some option other than `.force` but also have it in the `didBecomeActive` method in your AppDelegate.

Thanks!

**Comment by Lengo46** [comment]
Sure,
```

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions [UIApplicationLaunchOptionsKey: Any]?) -> Bool {

        //Siren Settings
        let siren = Siren.shared
        siren.debugEnabled = false
        siren.countryCode = "NL"
        siren.alertType = .option
        siren.checkVersion(checkType: .immediately)
        
        return true
    }

    func applicationWillEnterForeground(_ application: UIApplication) {
        Siren.shared.checkVersion(checkType: .immediately)
    }

    func applicationDidBecomeActive(_ application: UIApplication) {
        Siren.shared.checkVersion(checkType: .daily)
    }

```

**Comment by ArtSabintsev** [comment]
Get rid of this line

```
    func applicationWillEnterForeground(_ application: UIApplication) {
        Siren.shared.checkVersion(checkType: .immediately)
    }
```

and you should be fine.

Closing the issue, but feel free to respond.

**Comment by Lengo46** [comment]
Thanks for the help, but I still get the alert on a real device, although this version is running higher.

**Comment by Lengo46** [comment]
I've updated the appstore version to 1.3 (to make sure it isn't the naming), still doesn't work. Please reopen.

**Comment by ArtSabintsev** [comment]
I need some more information.

What's the bundle identifier for your app?

**Comment by Lengo46** [comment]
com.metersenweters.streetholder

**Comment by ArtSabintsev** [comment]
Everything seems to work fine for me on my device when I use your bundle ID for the Streetholder app.

The current version of your app in the store is 1.3.1.

When I change the version of the app to **1.3** and build to the device, I **DO** get the alert.
When I change the version of the app to **1.3.1** and build to the device, I **DO NOT** get the alert.
When I change the version of the app to **1.4.0** and build to the device, I **DO NOT** get the alert.

This is the expected behavior.

**Comment by Lengo46** [comment]
I think i'm doing something wrong because I have different outcome:

When I change the version of the app to 1.2 and build to the device, I DO get the alert.
When I change the version of the app to 1.3 and build to the device, I DO get the alert.
When I change the version of the app to 1.3.1 and build to the device, I DO get the alert.
When I change the version of the app to 1.4.0 and build to the device, I DO get the alert.

Basically i'm always getting the alert. I'm using a iPhone 6s.

**Comment by ArtSabintsev** [comment]
I used the code in the sample app, and it worked as is. 

I just replaced the bundle ID with yours and the countryCode with yours. 

Can you try doing that as well? Does the problem still persist in the sample app?

**Comment by Lengo46** [comment]
Found the problem. Framework was looking at the (wrong) localized CFBundleShortVersionString.

**Comment by ArtSabintsev** [comment]
How did that happen? Does a PR need to be made to fix this issue for others?

**Comment by Lengo46** [comment]
No it was my error, I had a hard-coded version number inside my InfoPlist.strings (localization file) for the Dutch language. The simulator was running EN, that why my real-device kept giving the error.

**Comment by ArtSabintsev** [comment]
My pleasure. Thanks for following up!
